Upmerge 05.03.2021

anangl opened the pull request:
Update the following projects:
- zephyr (see https://github.com/nrfconnect/sdk-zephyr/pull/479)
- trusted-firmware-m (see https://github.com/nrfconnect/sdk-trusted-firmware-m/pull/8)
- mcuboot (see https://github.com/nrfconnect/sdk-mcuboot/pull/145)

to align with https://github.com/zephyrproject-rtos/zephyr/commit/7074254937ae20f90174309b930ed8edfb53666e.

carlescufi commented:
@gmarull could you please take a look at the doc build error?

```
Error: /home/runner/work/sdk-nrf/sdk-nrf/ncs/_build/zephyr/rst/doc/reference/networking/sockets.rst:4: WARNING: Duplicate ID: "group__bsd__sockets_1gab9d2f7ca5c94bd51cdab3e1913b66e2d".
/home/runner/work/sdk-nrf/sdk-nrf/ncs/_build/zephyr/rst/doc/reference/networking/sockets.rst:4: WARNING: Duplicate explicit target name: "group__bsd__sockets_1gab9d2f7ca5c94bd51cdab3e1913b66e2d".
```

gmarull commented:
I see multiple `@defgroup bsd_sockets` in Zephyr, the last one introduced recently in `include/net/socket_types.h` by @hubertmis. I'll checkout and see.

gmarull commented:
I've just pulled and built docs for this branch and I get no errors on a clean build
```
(.venv) ➜  doc git:(upmerge_20210305) cmake -GNinja -B_build .
(.venv) ➜  doc git:(upmerge_20210305) ninja -C _build build-all 
...
(.venv) ➜  doc git:(upmerge_20210305) echo $?
0
```

mbolivar-nordic commented on samples/bluetooth/mesh/light_ctrl/src/lc_pwm_led.c:
```diff
@@ -23,11 +23,11 @@
 #else
 #error "Unsupported board: pwm-led0 devicetree alias is not defined"
-#define PWM_LABEL	""
+#define PWM_CTLR	""
```
You can use `DT_INVALID_NODE` instead of using an empty string as a node identifier. You could also just remove the PWM_CTLR define altogether here. The below `DEVICE_DT_GET` would fail at link time, but the program will never link because of the `#error` on line 29, so it's not an issue. Your call.

anangl replied:
Fixed.

mbolivar-nordic reviewed (commented):
Nice work. I found one minor issue in a DEVICE_DT_GET conversion, but otherwise LGTM.

NordicBuilder commented:
You can find the documentation preview for this PR at this <a href='http://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-4109
'>link</a>. It will be updated about 10 minutes after the documentation build succeeds.

*Note: This comment is automatically posted by the Documentation Publishing GitHub Action.*

krbvroc1 commented:
http://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-4109/nrf/releases/release-notes-latest.html

The release notes provides a Zephyr commit and a comment that it is still based on v2.4.99. I thought this was updating to 2.5.99?

ru-fu commented:
> 
> 
> http://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-4109/nrf/releases/release-notes-latest.html
> 
> The release notes provides a Zephyr commit and a comment that it is still based on v2.4.99. I thought this was updating to 2.5.99?

Yes, this must be updated. Also in nrf/doc/versions.json.

ru-fu commented on doc/nrf/known_issues.rst:
```diff
@@ -374,7 +374,7 @@ NCSDK-8223: GATT requests might deadlock RX thread
   The deadlock is resolved by a 30 second time-out, but the ATT bearer cannot transmit without reconnecting.
 
-  **Workaround:** Set :option:`CONFIG_BT_L2CAP_TX_BUF_COUNT` >= :option:`CONFIG_BT_ATT_TX_MAX` + 2.
+  **Workaround:** Set :option:`CONFIG_BT_L2CAP_TX_BUF_COUNT` >= ``CONFIG_BT_ATT_TX_MAX`` + 2.
```
@joerchan Does this workaround still make sense if the option does not exist anymore?

joerchan replied:
The workaround makes sense for older releases. We are pulling in the fix now.

pszkotak commented:
We observe an issue with the CoAP Client MTD Multiprotocol sample on nRF53.
Communication hangs after a few quick 'm' messages over NUS/BLE.
I do not observe this issue on the current NCS master.

pszkotak reviewed (changes requested):
We observe an issue with the CoAP Client MTD Multiprotocol sample on nRF53.
Communication hangs after a few quick 'm' messages over NUS/BLE.
I do not observe this issue on the current NCS master.

hubertmis commented:
> We observe an issue with the CoAP Client MTD Multiprotocol sample on nRF53.

Do you observe this issue only on the upmerge branch, or the NCS master branch as well?

hakonfam commented:
@tejlmand 

I see this when building `application_update` sample for nrf91ns

```
     **********************************************************************
     *
     *                    WARNING
     *
     * CMake variable BOARD set to "nrf9160dk_nrf9160" in:
     *     /home/haam/ncs/nrf/modules/mcuboot/CMakeLists.txt
     *
     * This is too late to make changes! The change was ignored.
     *
     * Hint: BOARD must be set before calling find_package(Zephyr ...).
     *
     **********************************************************************
```

MarekPieta reviewed (changes requested):
ZDebugMCUBootQSPI configuration does not build for nrf52840dk_nrf52840 (nrf_desktop). I am investigating this one.

MarekPorwisz commented:
> > We observe an issue with the CoAP Client MTD Multiprotocol sample on nRF53.
> 
> Do you observe this issue only on the upmerge branch, or the NCS master branch as well?

We do observe similar but not the same thing on master. On master I do observe an issue, that I was only able to reproduce with older phone(Android 6). The issue reported by Piotr does not depend on the android/BT version used and I think that Piotr has a network core assert, while our investigation of master issue shows there is no clear assert on net core, but app core assters due to incorrect/lack of response. @pszkotak could you confirm?

pszkotak commented:
> We do observe similar but not the same thing on master. On master I do observe an issue, that I was only able to reproduce with older phone(Android 6). The issue reported by Piotr does not depend on the android/BT version used and I think that Piotr has a network core assert, while our investigation of master issue shows there is no clear assert on net core, but app core assters due to incorrect/lack of response. @pszkotak could you confirm?

Yes, this looks like a completely different issue.
In my case I just conenct with the nRF Toolbox from my Android 10 device to the CoAP Client MTD Multiprotocol (at this state it is already conneted to the CoAP Server over Thread), and quickly press the button sending 'm' over NUS.
Then I see this on the net core:

```
Breakpoint 1, m_assert_handler (
    file=0x2100c92c <signal_thread_stack+916> "107", line=245)
    at /home/pisz/repo/ncs/zephyr/include/logging/log_core.h:463
463        return ((uint8_t *)data - (uint8_t *)__log_const_start)/
(gdb) bt
#0  m_assert_handler (file=0x2100c92c <signal_thread_stack+916> "107", line=245)
    at /home/pisz/repo/ncs/zephyr/include/logging/log_core.h:463
#1  0x0100423e in sym_S2UAPMFVIQXDUOA6CV7GJMB33TYHEUH5D6LHO5Q ()
#2  0x010045e6 in sym_RCMMPSTF2EML32JQ6R3HKP4RKIKXPCMPS23XHTA ()
#3  0x010004c0 in sym_UTQHJYLRNSDKNFRC6462JAV7TLCCKTYFC52KLPA ()
#4  0x010043bc in sym_VWGRUQT3LYZHLYS5KJ7UOCJEHLDJTWH4RVJBGKQ ()
#5  0x010266a6 in signal_thread (p1=<optimized out>, p2=<optimized out>, 
    p3=<optimized out>) at /home/pisz/repo/ncs/nrf/subsys/mpsl/mpsl_init.c:71
#6  0x01030950 in z_thread_entry (entry=0x102666d <signal_thread>, 
    p1=<optimized out>, p2=<optimized out>, p3=<optimized out>)
    at /home/pisz/repo/ncs/zephyr/lib/os/thread_entry.c:29
#7  0x01030950 in z_thread_entry (entry=0x102666d <signal_thread>, 
    p1=<optimized out>, p2=<optimized out>, p3=<optimized out>)
    at /home/pisz/repo/ncs/zephyr/lib/os/thread_entry.c:29
#8  0xe7fdcb0e in ?? ()```

MarekPieta commented:
I fixed the QSPI issues (needed to fix a few build warnings and reduce FLASH usage of MCUBoot bootloader in our configuration).

Still I observe problem with SRAM overflowed by 123 bytes on nrf52810dmouse (it seems like the memory consumption was increased and we are unable to fit). I will verify if I am able to reduce memory footprint here.

oyvindronningstad commented:
I think the following should fix https://jenkins-ncs.nordicsemi.no/job/latest/job/sub/job/test-fw-nrfconnect-tfm/job/master/337/console:

Add `CONFIG_TFM_PARTITION_AUDIT_LOG=n`

in samples/tfm_integration/tfm_ipc/prj.conf

pszkotak commented:
@hubertmis, do you have any comments regarding the assert in the MPSL? Are there any actions taken?

hubertmis commented:
> @hubertmis, do you have any comments regarding the assert in the MPSL? Are there any actions taken?

I don't. @czeslawmakarski, do you?

mbolivar-nordic commented:
> I see this when building `application_update` sample for nrf91ns

@hakonfam the warning is real and indicates a bug in the application

mbolivar-nordic commented:
> I see this when building `application_update` sample for nrf91ns

@hakonfam the warning is real and indicates a bug in the application

hakonfam reviewed (changes requested):
Ref @mbolivar-nordic 's comment about warning indicating a real issue. Will investigate.

anangl commented:
> > I see this when building `application_update` sample for nrf91ns
> 
> @hakonfam the warning is real and indicates a bug in the application

@mbolivar-nordic @tejlmand
I get a similar warning when I use upstream Zephyr and build e.g. `hello_world` for `nrf9160dk_nrf9160@1.0.0`, but not when the target is `nrf9160dk_nrf9160`. Is it actually a problem of the application?
```
     **********************************************************************
     *
     *                    WARNING
     *
     * CMake variable BOARD set to "nrf9160dk_nrf9160" in:
     *     /home/andy/zephyrproject/zephyr/cmake/app/boilerplate.cmake
     *
     * This is too late to make changes! The change was ignored.
     *
     * Hint: BOARD must be set before calling find_package(Zephyr ...).
     *
     **********************************************************************
```

czeslawmakarski commented:
The bug with the mpsl is something we are aware of. It is indeed a regression, but not a blocker for the upmerge.

tejlmand commented:
> > > I see this when building `application_update` sample for nrf91ns
> > 
> > 
> > @hakonfam the warning is real and indicates a bug in the application
> 
> @mbolivar-nordic @tejlmand
> I get a similar warning when I use upstream Zephyr and build e.g. `hello_world` for `nrf9160dk_nrf9160@1.0.0`, but not when the target is `nrf9160dk_nrf9160`. Is it actually a problem of the application?

This happens because we reuse `zephyr_file()` to populate a list of Kconfig fragments for the board.
But as we need a list for the child image board, then we make a locate BOARD change which is restricted to local scope.
However, this also triggers the warning that was introduced here: https://github.com/zephyrproject-rtos/zephyr/pull/32556

Will look into the best approach to achieve what we want, while still keep the warning, cause in most situations setting the BOARD in CMake is an error, but not in this particular case.

Note, this is only a warning, and code still behaves correctly, so please continue testing.

anangl commented:
> Ref @mbolivar-nordic 's comment about warning indicating a real issue. Will investigate.

@hakonfam The warning should be gone now.

carlescufi commented:
@ioannisg @joerchan we need to fix this: https://github.com/zephyrproject-rtos/zephyr/issues/33381

anangl commented:
Rebased on master.

anangl commented:
Updated one more assignment to `CONFIG_LOG_MINIMAL` that was recently added in master. Replaced ineffective `CONFIG_LOG_MODE_MINIMAL=n` with `CONFIG_LOG_MODE_DEFERRED=y`, or removed if such assignments were unnecessary anyway.

anangl commented:
Rebased once more, to make the compliance check happy.

ktaborowski reviewed (commented):
Please align manifest with nrfconnect master and (for pr purposes) use https://github.com/nrfconnect/sdk-homekit/pull/65
```
    - name: homekit
      repo-path: sdk-homekit
      revision: pull/65/head
      groups:
      - homekit
```

anangl commented:
Rebased on master to get homekit related entries in the manifest.

shantha-14 reviewed (approved):
Looks OK from test perspective

anangl commented:
Updated revisions of the zephyr, mcuboot, trusted-firmware-m, and homekit modules with proper SHAs.
